filter.go

```diff
@@ -0,0 +1,54 @@
+package rtreego
+
+// Filter is an interface for filtering leaves during search. The parameters
+// should be treated as read-only. If refuse is true, the currenty entry will
+// not be added to the result set. If abort is true, the search is aborted and
+// the current result set will be returned. In case both of them are true,
+// the current value will be included in the returned result set.
+type Filter interface {
+	Filter(results []Spatial, object Spatial) (refuse, abort bool)
+}
+
+// ApplyFilters applies the given filters and returns their consensus.
+func applyFilters(results []Spatial, object Spatial, filters []Filter) (bool, bool) {
+	var refuse, abort bool
+	for _, f := range filters {
+		ref, abt := f.Filter(results, object)
+
+		if ref {
+			refuse = true
+		}
+
+		if abt {
+			abort = true
+		}
+
+		if refuse && abort {
```

Comment:
can you add a comment explaining why this case requires both refuse and abort for early break?